josh-ui/src/patterns.rs

```diff
@@ -0,0 +1,40 @@
+use crate::route::{AppAnchor, AppRoute};
+use yew::{html, Html};
+
+pub fn list(list: Vec<(AppRoute, Html)>) -> Html {
+    html! {
+        <div id="pathlist" class="dirmode loaded">
+        <table class="pathlist">
+        {
+            for list.iter().map( |elt| {
+                html!{
+                    <AppAnchor route={elt.0.clone()}>
+                    <tr><td>
+                    { elt.1.clone() }
+                    </td></tr>
+                    </AppAnchor>
+                }
+            })
+        }
+        </table>
+        </div>
+    }
+}
+
+pub fn path_with_note(path: &str, note: &str) -> Html {
+    html! {<>
+       {
+           if let Some(d) = std::path::Path::new(path).file_name() {
+               d.to_string_lossy().to_string()
+           } else {
+               path.to_string()
+           }
+       }{ "/" }
```

Comment:
This trailing slash should probably not appear everywhere, but only on directories.